librz/asm/arch/hexagon/hexagon.c

```diff
@@ -20,848 +26,394 @@
 #include "hexagon.h"
 #include "hexagon_insn.h"
 #include "hexagon_arch.h"
+#include "hexagon_reg_tables.h"
+RZ_API ut32 hex_resolve_reg_enum_id(HexRegClass class, ut32 reg_num) {
```

Comment:
Missing empty line. Please also use a proper name prefix if it's API. Also brief doxygen description.

Reply:
Proper naming means `hexagon_...`? If yes, those are a lot of changes. I would like to move this to another PR then.

Reply:
Ah, also. Those are techincally `IPI`s. But because the analsysis part relies so heavily on the `asm` part, it needs to be `API` so linking works. So it is not "really" API. After RzArch it will have the visibility of IPI again.

Reply:
Ok. Then it's fine as is for now. Empty line is still necessary though...